### PR TITLE
JW-952 cleanup

### PR DESCRIPTION
**Pull request by mclapinski**

### CODE REVIEW

File: library/cpp/derived_stats_algo.h [hunk 1]
```diff
@@ -45,43 +45,50 @@ template <typename ElemT>
...
        
         // If sigma was low, and is still low, 
         // we don't need to send anything 
         if ((psigma_ < 0.5) && (psigma_ > -0.5) && 
             (sigma_ < 0.5) && (sigma_ > -0.5))
-            return false;
+            return DSR_SKIP;
         else
-	    return true;
+	    return DSR_OK;
```
Comment (m-kostrzewa) [inline library/cpp/derived_stats_algo.h]: 1 space missing here ;)

File: compiler/main.cc [hunk 2]
```diff
@@ -1210,43 +1207,36 @@ int main(int argc, char** argv) {
...
-		else {
-			fprintf(stderr, "!!! Unrecognized option: %s\n", arg);
-			usage();
-		     }
-	  arg = strtok_s(NULL, " ", &saveptr);
-#else
+      } else if ((strcmp(arg, "-o") == 0) || (strcmp(arg, "-out") == 0)) {
+        out_path_is_absolute = (strcmp(arg, "-out") == 0) ? true : false;
+		  
+		arg = argv[++i];
```
Comment (jablonskim) [inline compiler/main.cc:1213]: Please change tabs to spaces and remove tabs from line above.

File: library/cpp/protocol/TBinaryProtocol.tcc [hunk 3]
```diff
@@ -665,20 +657,12 @@ int32_t TBinaryProtocolT<Transport_>::readIPADDR(
   }
   if (version == AF_INET) {
     boost::asio::ip::address_v4::bytes_type ipv4;
-#ifdef _WINDOWS
     this->trans_->readAll(ipv4.data(), 4);
```
Comment (jablonskim) [inline library/cpp/protocol/TBinaryProtocol.tcc]: `readAll` needs pointer to non-const data. `boost:array::data()` returns pointer to const data. Please check if it works with `boost::array`.

File: library/cpp/protocol/TBinaryProtocol.tcc [hunk 4]
```diff
@@ -665,20 +657,12 @@ int32_t TBinaryProtocolT<Transport_>::readIPADDR(
...
-#else
-    this->trans_->readAll(ipv4.c_array(), 4);
-#endif
     ipaddress = boost::asio::ip::address_v4(ipv4);
     return ret+4;
   } else if (version == AF_INET6) {
     boost::asio::ip::address_v6::bytes_type ipv6;
-#ifdef _WINDOWS
     this->trans_->readAll(ipv6.data(), 16);
```
Comment (jablonskim) [inline library/cpp/protocol/TBinaryProtocol.tcc]: `readAll` needs pointer to non-const data. `boost:array::data()` returns pointer to const data. Please check if it works with `boost::array`.

**Review by jablonskim — CHANGES_REQUESTED** [review]

File: compiler/main.cc (hunk 2 above)
Comment (mclapinski) [reply compiler/main.cc:1213]: Dude, those aren't my changes. Must have been there before.

**Review by jablonskim — APPROVED** [review]

**Review by katrybacka — APPROVED** [review]

**Pull request merged**